### DISCUSSION TITLE
[Adonis 4.1]Decrypt cookie or session in socket.io middleware to access the user credentials

### DISCUSSION THREAD

**Discussion by tayfunakgc** [adonisjs/core#1408] [Help]
I have a Socket.io provider. 
Users connect to the socket in some pages and I can get their cookies easily. (socket.handshake.headers.cookie)
JSON output of cookie: 
```javascript
{
    'adonis-session': 'string value',
    'XSRF-TOKEN': 'string value',
    'adonis-session-values': 'string value',
    'io': 'string value'
}
```

But the problem is parsing the cookie to check the user.
I tried to decrypt the data. But it returns null; (I thought I can access with Encryption.decrypt())

So is there any way to do this? How can I check the user without HTTP ctx?

**Comment by thetutlage** [comment]
Which version of the framework?

**Reply by tayfunakgc** [reply]
Working on Adonis 4.1.0
